scripts/build-snarkyjs-node.sh

```diff
@@ -2,7 +2,12 @@ SNARKY_JS_PATH=$1
 [ -z "$SNARKY_JS_PATH" ] && SNARKY_JS_PATH=src/lib/snarky_js_bindings/snarkyjs
 
 dune b src/lib/crypto/kimchi_bindings/js/node_js \
-&& dune b src/lib/snarky_js_bindings/snarky_js_node.bc.js || exit 1
+&& dune b src/lib/snarky_js_bindings/snarky_js_node.bc.js src/lib/snarky_js_bindings/snarky_js_types.exe || exit 1
+
+# auto-generate js types from field derivers
+# todo declare these files as a dune target somehow
```

Comment:
See the rule in e.g. `/dune` for `graphql_schema.json` for how this is done

Reply:
Like with that, it would be good to use promotion so that we can see the results in-tree

Reply:
thanks - I made this work now!
https://github.com/MinaProtocol/mina/pull/10938/commits/fee40045a44a47788e437c13baee2208f71800fa
https://github.com/o1-labs/snarkyjs/pull/158/commits/1b287823b09d6a49f6ff764932a1d70007787a20